code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,165 @@
+/**
+	Time agent
+
+	You've got 5 minutes of playtime to do something weird
+
+	If you succeed, the stations threat level is escalated
+
+	If you fail, something weird happens.
+
+	Are you a bad enough dude to make sure a corgi, a rubber duck, and a bucket are in the same place at the same time?
+
+**/
+
+/datum/role/time_agent
+	name = TIMEAGENT
+	id = TIMEAGENT
+	required_pref = ROLE_MINOR
+	logo_state = "time-logo"
+	var/list/objects_to_delete = list()
+
+
+/datum/role/time_agent/Greet(var/greeting,var/custom)
```

Comment:
It would be great if we could think of a way to handle this greeting, give them 60 seconds to read it undisturbed, THEN warp them in

because boy we have some slow readers on /vg/ and 5 minutes means you do not want to waste a second

we should also consider a blob-overmind-esque timer on your screen so you can keep track of the time until the rift dies